page/renderer/ComponentUtils.js

```diff
@@ -0,0 +1,67 @@
+export default class ComponentUtils {
+  /**
+   * Initializes the registry used for managing component utils.
+   *
+   * @param {ObjectContainer} oc The application's dependency injector - the
+   *        object container.
+   */
+  constructor(oc) {
+    /**
+     * The application's dependency injector - the object container.
+     *
+     * @type {ObjectContainer}
+     */
+    this._oc = oc;
+
+    /**
+     * Map of registered utilities.
+     *
+     * @type {Object<string, Object>}
+     */
+    this._utilities = {};
+  }
+
+  /**
+   * Registers single utility class or multiple classes in alias->class mapping.
+   *
+   * @param {function(new: Object)|Object<string, function(new: Object)>} componentUtilityClass
+   * @param {string|null} alias
+   * @returns {Object|Object<string, Object>}
+   */
+  register(componentUtilityClass, alias = null) {
+    if (typeof componentUtilityClass === 'function') {
+      this._utilities[
+        alias || componentUtilityClass.name
+      ] = componentUtilityClass;
+    } else if (typeof componentUtilityClass === 'object') {
+      for (const alias of Object.keys(componentUtilityClass)) {
+        if (!componentUtilityClass.hasOwnProperty(alias)) {
+          continue;
+        }
+
+        this._utilities[alias] = componentUtilityClass[alias];
+      }
+    }
+  }
+
+  /**
+   * Returns object containing all registered utilities
+   *
+   * @returns {Object<string, Object>}
+   */
+  getUtils() {
```

Comment:
I would use some memoize cache for better performance impact.  It is useless creating new utilities object for every call.

Reply:
Created instance cache. When somebody calls `register` after the `getUtils`. A new utility is added and immediately instantiated.